v03_pipeline/lib/tasks/base/base_pipeline_task.py

```diff
@@ -0,0 +1,43 @@
+import hail as hl
+import luigi
+
+from v03_pipeline.lib.misc.io import write_ht
+from v03_pipeline.lib.model import DatasetType, Env, ReferenceGenome
+
+
+class BasePipelineTask(luigi.Task):
+    env = luigi.EnumParameter(enum=Env)
+    reference_genome = luigi.EnumParameter(enum=ReferenceGenome)
+    dataset_type = luigi.EnumParameter(enum=DatasetType)
+    hail_temp_dir = luigi.OptionalParameter(
+        default=None,
+        description='Networked temporary directory used by hail for temporary file storage. Must be a network-visible file path.',
+    )
+
+    def output(self) -> luigi.Target:
+        raise NotImplementedError
+
+    def complete(self) -> bool:
+        raise NotImplementedError
+
+    def run(self) -> None:
```

Comment:
This is the `run` method which each task will inherit from.  The general flow is for every task to define an `initialize_table` function, which will run if the output does not exist and then an `update` to actually perform the work.